pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -205,3 +243,32 @@ func (r *ReconcileCluster) getServiceAccountToken(instance metav1.Object) (strin
 
 	return "", fmt.Errorf("no matching secrets found")
 }
+
+func (r *ReconcileCluster) removeClusterFileFromTenant(clusterName string, tenantInfo types.NamespacedName, reqLogger logr.Logger) error {
+
+	tenantCR, err := r.getTenantCR(tenantInfo)
+	if err != nil {
+		return err
+	}
+
+	repo, _, err := manager.GetGitClient(tenantCR.Spec.GitRepoTemplate, tenantCR.GetNamespace(), reqLogger, r.client)
+	if err != nil {
+		return err
+	}
+
+	fileName := clusterName + ".yml"
+
+	if tenantCR.Spec.GitRepoTemplate.TemplateFiles == nil {
+		return nil
+	}
+
+	if _, ok := tenantCR.Spec.GitRepoTemplate.TemplateFiles[fileName]; ok {
+		delete(tenantCR.Spec.GitRepoTemplate.TemplateFiles, fileName)
+		err := r.client.Update(context.TODO(), tenantCR)
+		if err != nil {
+			return err
+		}
+	}
+
+	return repo.RemoveFile(fileName)
```

Comment:
I was able to produce race conditions the other way round. As the operator re-applied the already deleted file again, because it was in the map.

Reply:
Hmm makes sense. But like this if the `repo.RemoveFile` call fails, it won't be retried.
Anyways it's not really nice to call repo specific stuff from the cluster reconciler. Maybe we should introduce a "deleteMap"? Or set something like `tenantCR.Spec.GitRepoTemplate.TemplateFiles[fileName] = "remove"`?
And then let the repo reconciler handle the actual deletion.

Reply:
I had similar thoughts while implementing this, but found shortcomings for all my ideas. This one was the "best" one.

Got time for a zoom call today?

Reply:
As discussed in the zoom call:
* we agreed, that a more complex type to handle this would be the best solution, like:
```
'filename':
  content: 'somestring'
'filename2':
  delete: true
```
* we're at the point where we'd need validation and conversion webhook endpoints
* For the scope of this ticket we'll stick with the deletion magic string so we won't break the API
* I'll open follow-up issues to look into the webhooks

Reply:
Changed it to use the magic deletion string. After the file has been deleted it will also be removed from the map in the CRD.

I'm now using a type internally that's like the one we discussed in the zoom call. When we change the CRDs we can simplify the logic more.